webapp/app/templates/lotse/form_steuernummer.html

```diff
@@ -0,0 +1,185 @@
+
+
+{% block form_content %}
+    <div class="form-row-center">
+        {{ macros.render_field(form.steuernummer_exists, first_field=True if not render_info.form.errors) }}
+    </div>
+    <div class="form-row-center">
+        {{ macros.render_field(form.bundesland) }}
+    </div>
+
+    <div class="form-row-center">
+        {{ macros.render_field(form.bufa_nr) }}
+    </div>
+
+    <div class="form-row-center">
+        {{ macros.render_field(form.steuernummer) }}
+    </div>
+
+    <div class="additional-input-content" for="request_new_tax_number">
+        <h2 class="form-sub-heading-smaller">{{ _('form.lotse.request_new_tax_number.title') }}</h2>
+        {{ _('form.lotse.request_new_tax_number.intro') }}
+        {{ macros.render_field(form.request_new_tax_number) }}
+    </div>
+
+{% endblock %}
+
+{% block optional_js %}
+    {{ super() }}
+    <script src="{{ url_for('static', filename='js/field_utils.js') }}" type=text/javascript></script>
+    <script>
+        let taxOffices = JSON.parse('{{ render_info.additional_info['tax_offices']|tojson }}');
+        let TAX_NUMBER_FORM_STATES = {
+            'NothingSelected': 'NothingSelected',
+            'StateSelectionTaxNumberExists': 'StateSelectionTaxNumberExists',
+            'StateSelectionNoTaxNumberExists': 'StateSelectionNoTaxNumberExists',
+            'TaxOfficeSelection': 'TaxOfficeSelection',
+            'TaxNumberInput': 'TaxNumberInput',
+            'RequestNewTaxNumber': 'RequestNewTaxNumber'
+        }
+        let taxNumberFormState = TAX_NUMBER_FORM_STATES.NothingSelected;
+
+        document.addEventListener('DOMContentLoaded', () => {
+            // Register event handlers
+            document.querySelectorAll('input[type=radio][name=steuernummer_exists]').forEach((field)=>{
+                field.addEventListener("click",() => {changeTaxNumberExists(); runThroughAllStates(); handleState();});
+            });
+            document.getElementById('bundesland').addEventListener("change",() => {changeBundesland(); runThroughAllStates(); handleState();});
+            document.getElementById('bufa_nr').addEventListener("change",() => {changeTaxOffice(); runThroughAllStates(); handleState();});
+            // Check if input data should change state
+            runThroughAllStates();
+            assignAssociatedContent();
+            handleState();
+        });
+
+        function updateTaxOfficeList() {
+            let stateDropdown = document.getElementById("bundesland");
+            let bufaNrDropdown = document.getElementById("bufa_nr");
+            var selectedState = stateDropdown.options[stateDropdown.selectedIndex].value.toLowerCase();
+            var selectedTaxOffices = taxOffices.find(taxOffice => taxOffice.state_abbreviation === selectedState)
+            let oldBufaNrValue = bufaNrDropdown.value
+            deleteSelectOptions(bufaNrDropdown, defaultOptionText="{{ _('form.select_input.default_selection') }}");
```

Comment:
Why are some of these declared with `var` and some with `let`?

I'd prefer if we didn't mix "new"-style `let`/`const` with old-style `var`. Actually, I'd prefer if we didn't `var` at all unless there's a very very very very good reason for it.

Reply:
Yes, I overlooked that and I will change it to `let` and I agree with not using `var` anymore